compass/sign/RemoteSignatureSource.java

```diff
@@ -56,11 +57,28 @@ public RemoteSignatureSource(String uri,
    * @param uri the URI of the host to connect to
    */
   public RemoteSignatureSource(String uri) {
-    this(ManagedChannelBuilder.forTarget(uri).usePlaintext().idleTimeout(5, TimeUnit.SECONDS).build());
+    this.channelBuilder = getManagedChannelBuilder(uri);
+    this.serviceStub = SignatureSourceGrpc.newBlockingStub(channelBuilder.build());
+  }
+
+  private ManagedChannelBuilder getManagedChannelBuilder(String uri,
+                                                         String trustCertCollectionFilePath,
+                                                         String clientCertChainFilePath,
+                                                         String clientPrivateKeyFilePath) throws SSLException {
+    return NettyChannelBuilder
+      .forTarget(uri)
+      .idleTimeout(5, TimeUnit.SECONDS)
```

Comment:
I suppose you want it to match the security property

```suggestion
      .idleTimeout(Integer.valueOf(Security.getProperty("networkaddress.cache.ttl"))
, TimeUnit.SECONDS)
```

Reply:
I had the some thought: to have this * 2 the amount of cache.ttl property. If I apply this change though we definitely do not want the property to be configurable (e.g. the property defaults to "-1" which means to have the cache valid indefinetely). We will be multipliying here a negative number...

Reply:
Why *2?

Reply:
So we have a large enough window to make sure the cache expired for sure. That's all.